### ISSUE TITLE
if_attribute should works for time and between

### ISSUE THREAD

**Issue by kekrops** [stffn/declarative_authorization#11]
The if_attribute should support the following:
if_attribute :created_at => 1.month.ago..Time.now

**Comment by stffn** [comment]
Yes, this would be a nice feature.  I would be happy to help in its development and accept patches.

**Comment by seivan** [comment]
Yeah I was dealing with this today. We(camonznz on #ror) found a hack with has_permission on :comments, to [:actions] if_attribute 
                   :current_comment => { :created_at is_in Timespan }'

But what about doing 
        has_permission_on post :to => [:edit, update] do
            if_attribute :user => is {current_user} if post.created_at < 5.minutes.ago
        end

Or something similar.

**Comment by seivan** [comment]
And there is also a fork with 
http://github.com/trooster/declarative_authorization/commit/65659c6a576565664ba6a69660f1a31331a9ce5d

Where he uses gt, lt, gte and similar operators.
stffn, do you know how you can hack declarative_authorization to work with times now, I could really use it.

**Comment by CamonZ** [comment]
actually a possible implementation is adding [:custom, block] when the validation doesn't fit any of available methods in Authorization::Reader::AuthorizationRulesReader class. such that the rule could be:

```
if_attribute :current_associated_object => { :created_at =>  custom lambda { |value| boolean operation } }
```

where :current_associated_object would be the instance of a given object from an association say the object identified by :comment_id in /users/:user_id/comments/:comment_id

**Comment by stffn** [comment]
You can overwrite ObligationScope#attribute_value to handle time values (if it doesn't work by default).

**Comment by stffn** [comment]
@CamonZ such a custom block always has to work in both cases: a simple programmatic check like permitted_to? :read, object and for rewriting SQL queries.

**Comment by contentfree** [comment]
Is this issue fixed now? Doesn't the "is_in { someRange }" handle this? Aren't all of the if_attribute related to the current associated object?

Looks like you should be able to accomplish this with something like
    if_attribute :user => is {current_user}, 
                 :created_at => is_not_in { 5.minutes.ago..Time.now }

I guess I get what you're saying though... you want to do a comparison using a _different_ attribute value from the related object. I think that your example case is solvable with the current state of decl_auth.

**Comment by stffn** [comment]
Please try it.  There is no test for this kind of condition in the plugin's test suite.  If you'd like to contribute it, I'm glad to accept a patch.

**Comment by contentfree** [comment]
I was just saying that I think CamonZ' issue is covered by the existing codebase.

**Comment by aepstein** [comment]
@seivan - the gt, lt, gte, lte tests would be really helpful to have in official codebase.  Any reason this patch has not been incorporated?
